lib/ecto/query/builder.ex

```diff
@@ -536,14 +553,47 @@ defmodule Ecto.Query.Builder do
   """
   def keyword!(kw) do
     unless Keyword.keyword?(kw) do
```

Comment:
This can be written with a cond:

```elixir
cond do
  Keyword.keyword?(kw) ->
    kw
  is_binary(kw) ->
    raise ArgumentError, "...binary error message..."
  true ->
    raise ArgumentError, "...general error message..."
end
```

Reply:
Actually, I ignore this, I don't think we need two different error messages. Let me write a single message that handles both cases, one sec.